### PR TITLE
Change ESLint quote-props rule to as-needed

### PR DESCRIPTION
**Pull request by butterthebuddha**

### CODE REVIEW

**Comment by awesomeaniruddh** [comment]
@BerkeleyTrue Do you want me to fix the files ESLint is flagging in this PR?

**Comment by BerkeleyTrue** [comment]
@awesomeaniruddh Yes. That would be great!

File: server/utils/auth.js [hunk 1]
```diff
@@ -24,9 +24,9 @@ export function setProfileFromGithub(
   },
   {
     id: githubId,
-    'avatar_url': picture,
+    avatar_url: picture,
     email: githubEmail,
```
Comment (BerkeleyTrue) [inline server/utils/auth.js]: Do these come up as errors in ESlint?

Comment (awesomeaniruddh) [inline server/utils/auth.js]: @BerkeleyTrue Yep

Comment (BerkeleyTrue) [inline server/utils/auth.js]: Is there a version that would require props with hyphens and underscores to be quoted?

Comment (BerkeleyTrue) [inline server/utils/auth.js]: Looking at the docs, there doesn't seem to be. We'll go with this.

**Comment by BerkeleyTrue** [comment]
I'm surprised there weren't more lint errors. Excellent work @awesomeaniruddh

**Comment by awesomeaniruddh** [comment]
Hyphens are required to be quoted but undercores are not and a rule for them doesn't seem to exist. Thanks, wasn't really a lot of work :)

**Pull request merged**